### PR TITLE
Adding K2 Intro Tutorials for Jahia Release

### PR DESCRIPTION
**Pull request by scfleming**
Would be good if this could be merged with master branch by end of week of Feb. 4th.

### CODE REVIEW

File: notebooks/MAST/K2/K2_Lightcurve/k2_lightcurve.ipynb [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+   "source": [
+    "fits.info(fits_file)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's examine the binary table in the first FITS extension, since that contains the arrays of timestamps and fluxes we want to plot.  We will use the astropy.fits `getdata` function to access the table from the first extension HDU, and then show the columns of the table.  We can see included in the table are columns for the timestamps in Kepler BJD format (**TIME**), SAP flux (**SAP_FLUX**), and PDCSAP flux (**PDCSAP_FLUX**)."
+   ]
```
Comment (ceb8) [inline notebooks/MAST/K2/K2_Lightcurve/k2_lightcurve.ipynb]: "We can see included in the table are columns for the timestamps" -> "We can see that the table contains columns for the timestamps". Maybe more clear? I'm not sure, but something about this sentence seemed awkward to me.

File: notebooks/MAST/K2/K2_Lightcurve/k2_lightcurve.ipynb [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Start figure and axis.\n",
+    "fig, ax = plt.subplots()\n",
```
Comment (ceb8) [inline notebooks/MAST/K2/K2_Lightcurve/k2_lightcurve.ipynb]: Add `figsize=(12,8)` argument? Or whatever size you want, just maybe bigger than the default.

File: notebooks/MAST/K2/K2_Lightcurve/k2_lightcurve.ipynb [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
```
Comment (ceb8) [inline notebooks/MAST/K2/K2_Lightcurve/k2_lightcurve.ipynb]: Suggesting making this plot larger too (`figsize=(12,8)`?)

File: notebooks/MAST/K2/K2_Lightcurve/k2_lightcurve.ipynb [hunk 4]
```diff
@@ -0,0 +1,334 @@
...
+    "ax.set_ylabel(\"PDCSAP Flux (e-/s)\")\n",
+    "ax.set_xlabel(\"Time (TBJD)\")\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Intringuingly, some of the largest outliers in the positive flux direction are flagged: are these bad measurements that should be excised from the time series?  Finding out the quality flag value and converting the value to its consitutent bit masks to understand why these points are flagged would be the first step.  We encourage you to do this as a follow-up to this tutorial."
```
Comment (ceb8) [inline notebooks/MAST/K2/K2_Lightcurve/k2_lightcurve.ipynb]: "Intringuingly" I'm not sure if you meant "interestingly" or "intriguingly."

File: notebooks/MAST/K2/K2_Lightcurve/k2_lightcurve.ipynb [hunk 5]
```diff
@@ -0,0 +1,334 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Start figure and axis.\n",
+    "fig, ax = plt.subplots()\n",
```
Comment (ceb8) [inline notebooks/MAST/K2/K2_Lightcurve/k2_lightcurve.ipynb]: `figsize=(12,8)` again?

File: notebooks/MAST/K2/K2_Lightcurve/k2_lightcurve.ipynb [hunk 6]
```diff
@@ -0,0 +1,334 @@
...
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Understanding The Aperture Pixel Values\n",
+    "\n",
+    "We see the pixel values are integers, but what do they mean?  The pixels are bitmasks that encode information about each pixel.  You can find a summary of what the different values mean [here](https://archive.stsci.edu/kepler/manuals/archive_manual.pdf#page=20).  For example, a pixel in the aperture that might have a value of 15 can be broken down into power of 2 like: 8+4+2+1 = 15.  Referencing the table of values, this means this particular pixel was in was used to calculate the Pixel Response Function (PRF) centroid, was used to calculate the flux weighted centroid, was part of the optimal photometric aperture, and was collected by the spacecraft.  Numpy has a bulit-in function that can convert an integer into a binary bit mask.  Let's use that now one one of the common values we see in our displayed image above."
```
Comment (ceb8) [inline notebooks/MAST/K2/K2_Lightcurve/k2_lightcurve.ipynb]: "this particular pixel was in was used to calculate the Pixel" -> extra "was in"

"one one" -> "on one"

File: notebooks/MAST/K2/K2_Lightcurve/k2_lightcurve.ipynb (hunk 1 above)
Comment (scfleming) [reply notebooks/MAST/K2/K2_Lightcurve/k2_lightcurve.ipynb]: re-worded to say "We can see that the table includes columns for "

File: notebooks/MAST/K2/K2_Lightcurve/k2_lightcurve.ipynb (hunk 2 above)
Comment (scfleming) [reply notebooks/MAST/K2/K2_Lightcurve/k2_lightcurve.ipynb]: Sized adjusted.

File: notebooks/MAST/K2/K2_Lightcurve/k2_lightcurve.ipynb (hunk 3 above)
Comment (scfleming) [reply notebooks/MAST/K2/K2_Lightcurve/k2_lightcurve.ipynb]: Size adjusted.

File: notebooks/MAST/K2/K2_Lightcurve/k2_lightcurve.ipynb (hunk 4 above)
Comment (scfleming) [reply notebooks/MAST/K2/K2_Lightcurve/k2_lightcurve.ipynb]: typo fixed

File: notebooks/MAST/K2/K2_Lightcurve/k2_lightcurve.ipynb (hunk 5 above)
Comment (scfleming) [reply notebooks/MAST/K2/K2_Lightcurve/k2_lightcurve.ipynb]: Sized updated.

File: notebooks/MAST/K2/K2_Lightcurve/k2_lightcurve.ipynb (hunk 6 above)
Comment (scfleming) [reply notebooks/MAST/K2/K2_Lightcurve/k2_lightcurve.ipynb]: typos fixed.

File: notebooks/MAST/K2/K2_TPF/k2_tpf.ipynb [hunk 7]
```diff
@@ -0,0 +1,367 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Start figure and axis.\n",
+    "fig, ax = plt.subplots()\n",
+    "\n",
```
Comment (ceb8) [inline notebooks/MAST/K2/K2_TPF/k2_tpf.ipynb]: Suggest increasing figure size as in the other notebook. (`fig.set_size_inches(12., 8.)`)

File: notebooks/MAST/K2/K2_TPF/k2_tpf.ipynb [hunk 8]
```diff
@@ -0,0 +1,367 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Start figure and axis.\n",
+    "fig, ax = plt.subplots()\n",
```
Comment (ceb8) [inline notebooks/MAST/K2/K2_TPF/k2_tpf.ipynb]: Suggest increasing size here too (`fig.set_size_inches(12., 8.)`).

File: notebooks/MAST/K2/K2_TPF/k2_tpf.ipynb [hunk 9]
```diff
@@ -0,0 +1,367 @@
...
```
Comment (ceb8) [inline notebooks/MAST/K2/K2_TPF/k2_tpf.ipynb]: Suggest increasing size here too (`fig.set_size_inches(12., 8.)`).

File: notebooks/MAST/K2/K2_FFI/beginner_how_to_use_ffi.ipynb [hunk 10]
```diff
@@ -0,0 +1,221 @@
...
+    "fits_file = \"https://archive.stsci.edu/missions/k2/ffi/ktwo2015092174954-c04_ffi-cal.fits\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Understanding The FFI FITS File Structure\n",
+    "\n",
+    "K2 FFI FITS files contain a primary HDU with metadata stored in the header.  There are then 21 extensions that contain more metadata in the header and stores the full frame image for that module+channel.  Note that all 21 modules are included in the FFI FITS file, even if some of them do not have image data.  Let's examine the structure of the FITS file using the astropy.fits `info` function, which shows the FITS file format in more detail.\n"
```
Comment (ceb8) [inline notebooks/MAST/K2/K2_FFI/beginner_how_to_use_ffi.ipynb]: There appear to be 84 more extensions?

File: notebooks/MAST/K2/K2_FFI/beginner_how_to_use_ffi.ipynb [hunk 11]
```diff
@@ -0,0 +1,221 @@
...
...eqcArUTl8BaQgyAq0wxk4n1L1j/BzfO0I57yEsGsYwuL6E5FgCBJJ7zq1cBVCiVq5Lua/xACsJ7UMpSmjccm8fILbUzVsJ13YAWElAkhbUGtiHWGJd272xHoBRytCi4D5AHi0E1EPeP1o4HkCfjKUvi3l8Ytw7YvdHDx8/uOd1aHxmsia8//7uzrOXirg3hONYAixUwtzb3NsHiE2uAiAaoX0AAADgVQL7tEBgH9JIEdg/fXFvAoihhcnDgvsAzCWwD/TCOLlMMe8d+iNiWRveF9wvh2tEOI4lAAAA8NJN7gKAmWKtLHTtc1OvogQAAAArlRjYX/MmGiDO4gAWAgAIq4XgognXAG0T2Adgilh9gXsHtRO8BwCKIVcBAECBhPahJBq4Mmw2y/8Aac35uct1jZ2zXdcRAAoiwEjtSgzsQ6tiBOkBAHidhcBgHYF9AHrgHkVsnqkAANHJVZRBrgLqIVcBUAyhfaBvoRtEzebXNOcAAFUS2Kd2rYRYTHojBtd4AGLKGQwRSoHX6SvgZwL7AOSW8v7hXkVsS3qNVp7fAAA0Sa4iLrkKAOiG0D7UKPQg/NrntbZSXcpmprfmae3+9na8AACA4JZM+BJigeVOX/rYJgAAwEgIEoCehVwEBwAAVpOrWEeuIh65CgDoltA+lKa1Rq4kJTQsJdQQWsxmUKMJAJCcNzDTG4F9AAAAaJ+AIQBzxLhvWPSFFnnGAgBEJVcRTwnz80uoITS5CgBgENoHelBic1JiTXOl3ocWjhlhzTkfcv/ibs72necAZBQqsO8NyOTy/u5u1r9PPZlsbn0AAADAeiGDlwKXANTEfYsSeVYCAJBRifPxS6xpLrkKcpOrACiK0D7UKtTg49rn5B6QrVFDM1JDjS/lrjn39gEAGiWwT+1KD+xDi1zzAQDSEm6B+QT2AQDi8rwFAMhOruK6Gubf11DjS7lrzr19AOBVQvtQopobulLU1nzUUG9pTV1JtQAAMAyD8Cb5tBjYr6FG6hNqgRYAAOqgr4BwBPYBWMN9BAAAFpKrWK+2Ofc11CtXAQBccJu7AKBgNTa5NTccY+0lHvdSj2vJxwwAoCJCnPREaAUAAAD68PDxQ+4SAAAAAGhBjXPVS53/P0XJGYFSj2vJxwwAOiO0D7HFHJRvNusG1SXXtnSbLchx7C6p4biWdswAACoSKrB/+hLmc2Cu93
```
Comment (ceb8) [inline notebooks/MAST/K2/K2_FFI/beginner_how_to_use_ffi.ipynb]: The download happens when you do the .info command. Move this warning to that step?

File: notebooks/MAST/K2/K2_TPF/k2_tpf.ipynb (hunk 7 above)
Comment (scfleming) [reply notebooks/MAST/K2/K2_TPF/k2_tpf.ipynb]: Size updated.

File: notebooks/MAST/K2/K2_TPF/k2_tpf.ipynb (hunk 8 above)
Comment (scfleming) [reply notebooks/MAST/K2/K2_TPF/k2_tpf.ipynb]: Size updated.

File: notebooks/MAST/K2/K2_TPF/k2_tpf.ipynb (hunk 9 above)
Comment (scfleming) [reply notebooks/MAST/K2/K2_TPF/k2_tpf.ipynb]: Size updated.

File: notebooks/MAST/K2/K2_FFI/beginner_how_to_use_ffi.ipynb (hunk 10 above)
Comment (scfleming) [reply notebooks/MAST/K2/K2_FFI/beginner_how_to_use_ffi.ipynb]: Correct, edited to explain there are 84 extensions (21 modules times 4 channels per module).

File: notebooks/MAST/K2/K2_FFI/beginner_how_to_use_ffi.ipynb (hunk 11 above)
Comment (scfleming) [reply notebooks/MAST/K2/K2_FFI/beginner_how_to_use_ffi.ipynb]: Moved the warning up earlier in the notebook as suggested.

File: notebooks/MAST/K2/K2_FFI/beginner_how_to_use_ffi.ipynb [hunk 12]
```diff
@@ -0,0 +1,277 @@
...
...vCMexBKpkLuk8jhdQCrkKgGBucxcAzRgHFDFW99ls0g5YYq1QZNAFxDT32hXrmnQ6zasl9TV+idLrAwBYwAQy+FGvC7Fs3g3D6UvuKgD49N33V/9NqsnJDx8/TKoHSmERMgAgt96ChL3t70ivxFRjj6L36Fuv18oYHEuganIV15mXC8QkVwGQldA+hPbaACFWs1YyAyUAAGhSr+FOyDnJzAQ3WnEtJO8eA5BXqonAc8Me5//eZGUAAMjPuBzqk3Ih4/d3d55rdMi9IRzHEmiKXMWP5CoAALoitA+1sGIQQDixr6dzV4UDgEoIU9IKb6SEH6W4rs95m/35v01R2+bdvPoAWCfEmxnHz4g1edkbJAEA4LJeg4Qp93tKT5L6+6BXYq7xOYznK33o9d4Qg2MJMJFcBUA4chUAwQntA18zGAJqVPu1yy8QAaiAwD6tKCGwn/KNNpDL2jD8+PXuPwD1ixHs+PTd9yYxAwBAQj2Pv1Pt+5ze6fzf9vy9oXzv7+4E9xvm+hOOYwlQqNrnJgN9qv3aJVcBNEBoHwDoS6omzqpwAADFKSGwD6WIGYYP+fb605e4tW7eha0XoFaxJgbHfBNjrOC+N0hSAwuBQbncQ4DWCBHGt/bekWpRM71S3eY+7wjZcwjut8e9IRzHEgCArOQqAKIQ2oea5F4xyEAJKFEr16Xc13gAuMBbjmlBrYH9UuqAqWIE4MfPdD8CqEuKMEeqcArUTl8BaQgyAq0wxk4n1L1j/BzfO0I57yEsGsYwuL6E5FgCBJJ7zq1cBVCiVq5Lua/xACsJ7UMpSmjccm8fILbUzVsJ13YAWElAkhbUGtiHWGJd272xHoBRytCi4D5AHi0E1EPeP1o4HkCfjKUvi3l8Ytw7YvdHDx8/uOd1aHxmsia8//7uzrOXirg3hONYAixUwtzb3NsHiE2uAiAaoX0AAADgVQL7tEBgH9JIEdg/fXFvAoihhcnDgvsAzCWwD/TCOLlMMe8d+iNiWRveF9wvh2tEOI4lAAAA8NJN7gKAmWKtLHTtc1OvogQAAAArlRjYX/MmGiDO4gAWAgAIq4XgognXAG0T2Adgilh9gXsHtRO8BwCKIVcBAECBhPahJBq4Mmw2y/8Aac35uct1jZ2zXdcRAAoiwEjtSgzsQ6tiBOkBAHidhcBgHYF9AHrgHkVsnqkAANHJVZRBrgLqIVc
```
Comment (eteq) [inline notebooks/MAST/K2/K2_FFI/beginner_how_to_use_ffi.ipynb]:
```suggestion
    "![kepler_focal_plane_layout_channels_color.png](kepler_focal_plane_layout_channels_color.png)"
```
Is there some specific reason for the "attachment" part here?  If you look at this notebook you'll see that it now contains the data in this PNG image *inside* the notebook, which I don't think was intended, given that you've also checked the PNG into the repo.  If instead you leave off the `attachment` part, it just links to the file like a standard web page, which is also better for improving the load speed of the page.

**Comment by ceb8** [comment]
@eteq We  (me and @scfleming ) removed the attachment, but were not able to rebase efficiently, we'd like to squash and merge when this PR goes through (we would have had to squash most of the history anyway).

Additionally, we excluded all notebooks except these new K2 notebooks, and the tests pass, so the problem is the overall run time. For this one instance, could this PR be approved and you can change back the exclude list in the master branch? 

Moving forward we need to figure out a way to add new notebooks without running into this timing issue. Note  that we tried the attachment change by itself, and the tests still failed, so that was not the reason for travis failing.

**Comment by eteq** [comment]
After some slack discussion with @ceb8 I think the best thing to do re: the rebasing was to do the "ultimate squash" - e.g., copying the notebook elsewhere, making a new commit with just the final commit and then pushing that up as a replacement. I then also put up the exclude change @ceb8 mentioned in https://github.com/spacetelescope/notebooks/pull/53#issuecomment-465316618.  So if travis passes on the *first* commit we should back the second one out, but if it passes on the second one but not the first we can merge this as a temporary solution to the time-out.  Hopefully @obviousrebel and I will have a better solution to this ASAP though.

@scfleming @ceb8 - there were two reasons why the rebase was complex: 1) there were multiple notebooks in one PR and 2) several times it looks like you merged changes from another branch into this one.  To address 1 I updated the contributing guide to mention one-notebook-per-PR is a safer practice (https://github.com/spacetelescope/notebooks/commit/9423708c27e4d8d943eaa4a09f9f52801a3b5704), since I'm now realizing there was nowhere it said that was the guideline.  For 2 I think the best thing is to try to just avoid merging unless it's necessary to pull from master or something (although maybe doing 1 will help with this since I'm not clear on why those merges were in there).  Happy to chat more about this offline if you want some tips.

**Comment by eteq** [comment]
Fun times!  Turns out if you push too quickly *both* builds have the later commit because Travis needs some time to spin-up... re-doing the experiment above...

**Comment by eteq** [comment]
I was thinking to use this to try to get things working via `exclude_notebooks`... but I think it's probably best to just merge this and fix it later so the notebook is at least up.  Should actually have done that last week - sorry!

**Comment by ceb8** [comment]
Thanks @eteq!

**Pull request merged**